### ISSUE TITLE
Accessing missing Layout through indexer off of Select applicative throwing ValueMissingException

### ISSUE THREAD

**Issue by ahives** [phatboyg/Machete#69]
Accessing missing Layout through indexer off of Select applicative throwing ValueMissingException

```
var patientControlNumber = transactions
                .Select(x => x.PatientDetail)[0]
                .Select(x => x.PayerClaimControlNumber)[0]
                .Select(x => x.PatientControlNumber)
                .Select(x => x.ReferenceIdentificationQualifier);
```

If PatientDetail/PayerClaimControlNumber this will blow up and it shouldn't.

**Comment by ahives** [comment]
This will be fixed in next commit

**Comment by phatboyg** [comment]
Why wouldn't this blow up? You're accessing the first element of an empty list.

**Comment by ahives** [comment]
I think I didn't explain this correctly. I meant to say that I have changed MissingLayoutList to the same pattern as MissingValueList.

But to answer your question it will blow up if I were to do something like...

```
var patientControlNumber = transactions
                .Select(x => x.PatientDetail)[0]
                .Select(x => x.PayerClaimControlNumber)[0].Value;
```

Our current pattern is not to blow up on this...

```
var patientControlNumber = transactions
                .Select(x => x.PatientDetail)[0]
                .Select(x => x.PayerClaimControlNumber)[0];
```

This will be fixed for both MissingSegmentList and MissingLayoutList.

**Comment by ahives** [comment]
this is fixed as of v1.0.245-develop
